Get hold of the real gmp.h, not the obsolete one ghc brings with
it. This gives sparc a chance

--- ../ghc-4.08.2/ghc/includes/Makefile.orig	Tue Aug  8 02:00:58 2000
+++ ../ghc-4.08.2/ghc/includes/Makefile	Tue Aug  5 10:25:58 2003
@@ -20,8 +20,8 @@
 all :: $(H_CONFIG) NativeDefs.h
 
 # gmp.h is copied from the GMP directory
-gmp.h : $(FPTOOLS_TOP)/ghc/rts/gmp/gmp.h
-	$(CP) $< $@
+gmp.h :
+	$(CP) $(LOCALBASE)/include/gmp.h $@
 
 # The fptools configure script creates the configuration header file 
 # and puts it in fptools/mk/config.h. We copy it down to here, prepending
